### PR TITLE
feat(governance): datazone MSK

### PR DESCRIPTION
**Pull request by vgkowski**
**Issue #, if available:**

## Description of changes:

Implement MSK topics integration in DataZone. Provides:
1. DataZone custom asset type for MSK topics
2. Custom authorizer granting MSK topics read access when subcription is granted in DataZone

**Checklist**

 

* [ ] Update tests
* [ ] Update docs
* [ ] PR title follows [conventional commit semantics](https://www.conventionalcommits.org/en/v1.0.0/) (`fix: `, `feat: `, `docs: `, ...)

## Breaking change checklist

**RFC issue #**:

* [ ] Migration process documented
* [ ] Implement warnings (if it can live side by side)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: framework/src/governance/lib/datazone-authorizer/resources/msk-producer-authorizer/index.mjs [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    const detail = event.detail
+
+    if (detail.eventType === "SUBSCRIBED") {
+        const asset = detail.asset
+        const environments = detail.environments
+
+        const kafkaForm = asset.formOutput.find((form) => form.typeName === "KafkaForm" )
+        const { clusterArn } = JSON.parse(kafkaForm.content)
+
+        await client.send(new UpdateConnectivityCommand({
```
Comment (lmouhib) [inline framework/src/governance/lib/datazone-authorizer/resources/msk-producer-authorizer/index.mjs:13]: Why do we have this here? We should not just trigger the action without verifying if its was successfully applied. In addition it mutate the infrastructure (the msk cluster), ideally we should have the user perform this action outside the authorizer.

File: framework/src/governance/lib/datazone-authorizer/resources/msk-producer-authorizer/index.mjs [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        }))
+
+        const environmentAccountIds = environments.map((e) => e.awsAccountId)
+        const kafkaClusterPolicy = JSON.stringify({
+            "Version": "2012-10-17",
+            "Statement": [
+                {
+                    "Effect": "Allow",
+                    "Principal": {
+                        "AWS": environmentAccountIds
```
Comment (lmouhib) [inline framework/src/governance/lib/datazone-authorizer/resources/msk-producer-authorizer/index.mjs:35]: Ideally this should be scoped down to a role and not be open wide in the account.

File: framework/src/governance/lib/datazone-msk-environment-authorizer.ts [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+      ],
+      inlinePolicies: {
+        IamPermissions: new PolicyDocument({
+          statements: [
+            new PolicyStatement({
+              effect: Effect.ALLOW,
+              actions: [
+                'iam:PutRolePolicy',
+              ],
+              resources: ['*'],
```
Comment (lmouhib) [inline framework/src/governance/lib/datazone-msk-environment-authorizer.ts:46]: We need to scope this,  and not leave it wide to only roles which are meant to be updated by DSF as specified by the user.

File: framework/src/governance/lib/resources/datazone-msk-authorizer-grant/index.mjs [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+
+function getIamResources(region, account, clusterName, clusterUuid, topic) {
+  return [
+    `arn:aws:kafka:${region}:${account}:topic/${clusterName}/${clusterUuid}/${topic}`,
+    `arn:aws:kafka:${region}:${account}:cluster/${clusterName}/${clusterUuid}`,
+    `arn:aws:kafka:${region}:${account}:group/${clusterName}/${clusterUuid}/*`
+  ]
+}
+
+const iamActions = [
```
Comment (lmouhib) [inline framework/src/governance/lib/resources/datazone-msk-authorizer-grant/index.mjs:14]: Lets rename this to the `readAction` for readability and comprehension.

**Review by lmouhib — CHANGES_REQUESTED** [review]

File: framework/src/governance/lib/datazone/datazone-msk-custom-asset-props.ts [hunk 5]
```diff
@@ -0,0 +1,18 @@
+import { RemovalPolicy } from 'aws-cdk-lib';
+
+export interface DataZoneMSKCustomAssetProps {
+  topicName: string;
```
Comment (lmouhib) [inline framework/src/governance/lib/datazone/datazone-msk-custom-asset-props.ts:4]: These should be readonly.

File: framework/src/governance/lib/datazone/datazone-msk-custom-asset.ts [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+              effect: Effect.ALLOW,
+              actions: [
+                'datazone:CreateAsset',
+                'datazone:CreateAssetType',
+                'datazone:CreateFormType',
+                'datazone:GetAssetType',
+                'datazone:GetFormType',
+                'glue:GetSchemaVersion',
+                'glue:ListSchemaVersions',
+                'datazone:CreateAssetRevision',
```
Comment (lmouhib) [inline framework/src/governance/lib/datazone/datazone-msk-custom-asset.ts:72]: Can you separate the Read action and Write action? Can you verify that Write action can be scoped down by requiring a tag attached to them?

File: framework/src/governance/lib/datazone/glue-schema-registry-version.ts [hunk 7]
```diff
@@ -0,0 +1,137 @@
...
+
+    if (props.schemaVersionNumber === undefined && props.latestVersion === undefined) {
+      throw new Error('Either schemaVersionNumber or latestVersion must be provided.');
+    }
+
+    this.removalPolicy = Context.revertRemovalPolicy(scope, props?.removalPolicy);
+    const handlerRole = new Role(this, 'HandlerRole', {
+      assumedBy: new ServicePrincipal('lambda.amazonaws.com'),
+      managedPolicies: [
+        ManagedPolicy.fromAwsManagedPolicyName('service-role/AWSLambdaBasicExecutionRole'),
```
Comment (lmouhib) [inline framework/src/governance/lib/datazone/glue-schema-registry-version.ts:67]: @FranMorilloAWS  please use a policy created within the DSF and not the basic execution role.

**Review by lmouhib — CHANGES_REQUESTED** [review]

File: framework/src/governance/lib/datazone-authorizer/resources/msk-producer-authorizer/index.mjs (hunk 1 above)
Comment (vgkowski) [reply framework/src/governance/lib/datazone-authorizer/resources/msk-producer-authorizer/index.mjs]: file removed

File: framework/src/governance/lib/datazone-authorizer/resources/msk-producer-authorizer/index.mjs (hunk 2 above)
Comment (vgkowski) [reply framework/src/governance/lib/datazone-authorizer/resources/msk-producer-authorizer/index.mjs]: file removed

File: framework/src/governance/lib/datazone-msk-environment-authorizer.ts (hunk 3 above)
Comment (vgkowski) [reply framework/src/governance/lib/datazone-msk-environment-authorizer.ts]: file removed

File: framework/src/governance/lib/resources/datazone-msk-authorizer-grant/index.mjs (hunk 4 above)
Comment (vgkowski) [reply framework/src/governance/lib/resources/datazone-msk-authorizer-grant/index.mjs]: done

**Pull request closed without merging**